src/client/clientlauncher.cpp

```diff
@@ -53,17 +53,13 @@ MainGameCallback *g_gamecallback = NULL;
 
 ClientLauncher::~ClientLauncher()
 {
-	if (receiver)
-		delete receiver;
+	delete receiver;
 
-	if (input)
-		delete input;
+	delete input;
 
-	if (g_fontengine)
-		delete g_fontengine;
+	delete g_fontengine;
 
-	if (device)
```

Comment:
Seems to be a method inherited by tons of things. Official documentation doesn't use the check, however the actual source files (eg. [source/Irrlicht/CIrrDeviceStub.cpp](https://sourceforge.net/p/irrlicht/code/HEAD/tree/trunk/source/Irrlicht/CIrrDeviceStub.cpp)) seem to use both at random:

```cpp
CIrrDeviceStub::~CIrrDeviceStub()
{
	VideoModeList->drop();

	if (GUIEnvironment)
		GUIEnvironment->drop();

	if (SceneManager)
		SceneManager->drop();
	
	if (VideoDriver)
		VideoDriver->drop();

	if (ContextManager)
		ContextManager->drop();

	if ( FileSystem )
		FileSystem->drop();

	if (InputReceivingSceneManager)
		InputReceivingSceneManager->drop();

	if (CursorControl)
		CursorControl->drop();

	if (Operator)
		Operator->drop();

	if (Randomizer)
		Randomizer->drop();

	CursorControl = 0;

	if (Timer)
		Timer->drop();

	if (Logger->drop())
		os::Printer::Logger = 0;
}
```

**irr::IrrlichtDevice** derives from **irr::IReferenceCounted**

Documentation for drop():

> bool **drop ()** const
> Drops the object. Decrements the reference counter by one. 

> When you create an object in the Irrlicht engine, calling a method which starts with 'create', an object is created, and you get a pointer to the new object. If you no longer need the object, you have to call drop(). This will destroy the object, if grab() was not called in another part of you program, because this part still needs the object. Note, that you only need to call drop() to the object, if you created it, and the method had a 'create' in it.

> A simple example:
> 
> If you want to create a texture, you may want to call an imaginable method IDriver::createTexture. You call ITexture* texture = driver->createTexture(dimension2d<u32>(128, 128)); If you no longer need the texture, call texture->drop(). If you want to load a texture, you may want to call imaginable method IDriver::loadTexture. You do this like ITexture* texture = driver->loadTexture("example.jpg"); You will not have to drop the pointer to the loaded texture, because the name of the method does not start with 'create'. The texture is stored somewhere by the driver.

> **Returns**:
> True, if the object was deleted.